src/structures/MessageAttachment.js

```diff
@@ -43,6 +43,21 @@ class MessageAttachment {
     return this;
   }
 
+  /**
+   * Sets whether this attachment is a spoiler
+   * @param {boolean} [spoiler=true] Whether the attachment should be marked as a spoiler
+   * @returns {MessageAttachment} This attachment
+   */
+  setSpoiler(spoiler = true) {
+    if (spoiler === true && !this.name.startsWith('SPOILER_')) {
```

Comment:
i would personally go for this approach
```js
const isSpoiler = this.name.startsWith("SPOILER_");
if (spoiler != isSpoiler) {
  if (spoiler) {
    this.name = `SPOILER_${this.name}`;
  } else {
    this.name = this.name.slice(8);
    // or
    this.name = this.name.replace(/^SPOILER_/, '');
  }
}
return this;
```

or at the least putting the `this.name.startsWith(...);` in a variable instead of calling it twice, and using an else if statement below

Reply:
I agree and slice may actually be better

Reply:
We could also maake an `isSpoiler` getter, and flatten indentation a bit:

```js
if (spoiler === this.isSpoiler) return this;

this.name = spoiler
  ? `SPOILER_${this.name}`
  : this.name.slice('SPOILER_'.length);

return this;
```

Reply:
But this.spoiler already exists ?

Reply:
yeah, i put the simple `isSpoiler` in the suggested code because of the problem mentioned before of it requiring a URL, if thats fixed then the getter could be used instead

Reply:
Yeah fixing the getter would be the best approach like I suggested below this